### PR TITLE
new industrial boards namino rosso / namino arancio

### PR DESCRIPTION
**Pull request by a-sapuppo**
Namino Boards: Namino combines the industrial world with the world of makers by making digital analog inputs, outputs and compatible Arduino shields available on the same board.

CPU: ESP32-S3-WROOM-1U-N4R8
```

NAMINO ROSSO provides the following input/output lines:
8 PNP digital inputs (up to 24Vdc)
1 knob input and button
8 PNP digital outputs (up to 24Vdc) of which 1 for stepper motor command with total current 200mA
1 SPDT relay output of 5A
8 single ended analog inputs (0-10V or 0-20mA) or 4 thermometric (NTC, PTC, PT1000) or 4 thermocouples (J, K, T) or 2 load cells, software configurable
1 analog output jumper configurable: 0-10V or 0-20mA
ARDUINO UNO compatible shield interface 


NAMINO ARANCIO provides the following input/output lines:
8 PNP digital inputs (up to 24Vdc)
1 knob input and button
8 PNP digital outputs (up to 24Vdc) of which 1 for stepper motor command with total current 200mA
1 SPDT relay output of 5A
4 analog inputs (0-10V or 0-20mA) software configurable
1 input for NTC (10kΩ) or board temperature reading
1 analog output jumper configurable: 0-10V or 0-20mA
ARDUINO UNO compatible shield interface 
```

## Related links
[namino main information site](https://namino.cc/)
[namino rosso board](https://namino.cc/boards/namino-rosso/)
[namino arancio board](https://namino.cc/boards/namino-arancio/)

### CODE REVIEW

File: variants/namino_rosso/pins_arduino.h [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+#include <stdint.h>
+#include "soc/soc_caps.h"
+
+#define USB_VID 0x303a
+#define USB_PID 0x1001
+
+#define NAMINO_ROSSO_BOARD
+
+
```
Comment (P-R-O-C-H-Y) [inline variants/namino_rosso/pins_arduino.h:16]: Same here: 
You should add this part, which is missing. To have all functions working properly.
```suggestion
#define NUM_DIGITAL_PINS        SOC_GPIO_PIN_COUNT    // GPIO 0..48
#define NUM_ANALOG_INPUTS       20                    // GPIO 1..20
#define EXTERNAL_NUM_INTERRUPTS NUM_DIGITAL_PINS      // All GPIOs

#define analogInputToDigitalPin(p)  (((p)<NUM_ANALOG_INPUTS)?(analogChannelToDigitalPin(p)):-1)
#define digitalPinToInterrupt(p)    (((p)<NUM_DIGITAL_PINS)?(p):NOT_AN_INTERRUPT)
#define digitalPinHasPWM(p)         (p < NUM_DIGITAL_PINS)
```

File: variants/namino_arancio/pins_arduino.h [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+#include <stdint.h>
+#include "soc/soc_caps.h"
+
+#define USB_VID 0x303a
+#define USB_PID 0x1001
+
+#define NAMINO_ARANCIO_BOARD
+
+
```
Comment (P-R-O-C-H-Y) [inline variants/namino_arancio/pins_arduino.h:16]: You should add this part, which is missing. To have all functions working properly.
```suggestion
#define NUM_DIGITAL_PINS        SOC_GPIO_PIN_COUNT    // GPIO 0..48
#define NUM_ANALOG_INPUTS       20                    // GPIO 1..20
#define EXTERNAL_NUM_INTERRUPTS NUM_DIGITAL_PINS      // All GPIOs

#define analogInputToDigitalPin(p)  (((p)<NUM_ANALOG_INPUTS)?(analogChannelToDigitalPin(p)):-1)
#define digitalPinToInterrupt(p)    (((p)<NUM_DIGITAL_PINS)?(p):NOT_AN_INTERRUPT)
#define digitalPinHasPWM(p)         (p < NUM_DIGITAL_PINS)
```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
PTAL on my comments, after resolving it we can merge :)

File: variants/namino_rosso/pins_arduino.h (hunk 1 above)
Comment (a-sapuppo) [reply variants/namino_rosso/pins_arduino.h:22]: added

File: variants/namino_arancio/pins_arduino.h (hunk 2 above)
Comment (a-sapuppo) [reply variants/namino_arancio/pins_arduino.h:22]: added

**Review by P-R-O-C-H-Y — APPROVED** [review]
LGTM

**Pull request merged**